pinc/misc.inc

```diff
@@ -732,6 +732,30 @@ function extract_zip_to(string $path_to_zip, string $output_directory): bool
     // No need to check the result of the function since it is checked in is_valid_zip_file
     $zip->open($path_to_zip);
 
+    if($remove_top_dir)
+    {
+        // find if there is a common prefix implying a common directory, remove it
+        $filenames = [];
+        $last_index = $zip->numFiles - 1;
+        for ($i = 0; $i <= $last_index; $i++)
+        {
+            $filenames[] = $zip->getNameIndex($i);
+        }
+        sort($filenames);
+        $prefix = $filenames[0];
```

Comment:
Since `$prefix` here is a base directory, `$basedir` would make more sense and help make the rest of the logic clearer.